test/information_estimates_test.cpp

```diff
@@ -0,0 +1,97 @@
+
+#ifndef SLAM_TOOLBOX_SLAM_TOOLBOX_INFORMATION_ESTIMATES_TEST_H_
+#define SLAM_TOOLBOX_SLAM_TOOLBOX_INFORMATION_ESTIMATES_TEST_H_
+
+#include <gtest/gtest.h>
+#include "slam_toolbox/experimental/information_estimates.hpp"
+
+
+TEST(UtilsInformationEstimatesTests, SigNumTest)
+{
+    InformationEstimates information_estimates;
+
+    ASSERT_EQ(utils::grid_operations::signum(10), 1) << "FAIL in positive";
+    ASSERT_EQ(utils::grid_operations::signum(-10), -1) << "FAIL in negative";
+    ASSERT_EQ(utils::grid_operations::signum(0), 0) << "FAIL in zero";
+}
+
+TEST(UtilsInformationEstimatesTests, RayCastTest)
+{
+    InformationEstimates information_estimates;
+
+    karto::Vector2<int> initial_pt{22,43};
+    karto::Vector2<int> final_pt{29,38};
+
+    std::vector<karto::Vector2<int>> cells = utils::grid_operations::rayCasting(initial_pt, final_pt);
+
+    ASSERT_EQ(cells[2].GetX(), 25) << "FAIL in X position for cell 3";
+    ASSERT_EQ(cells[2].GetY(), 41) << "FAIL in Y position for cell 3";
+
+    ASSERT_EQ(cells[4].GetX(), 27) << "FAIL in X position for cell 5";
+    ASSERT_EQ(cells[4].GetY(), 39) << "FAIL in Y position for cell 5";
+}
+
+TEST(UtilsInformationEstimatesTests, IntersectionPointsTest)
+{
+    karto::Vector2<kt_double> laser_start{1.5, 1.6};
+    karto::Vector2<kt_double> laser_end{6.1, 8.4};
+    std::vector<kt_double> cell_start{3.6, 8.2};
+    std::vector<kt_double> cell_end{4.7, 1.6};
+
+    std::vector<kt_double> int_points = utils::grid_operations::calculateCellIntersectionPoints(laser_start, laser_end, cell_start, cell_end);
+    
+    ASSERT_FLOAT_EQ(int_points[0], 4.06744) << "FAIL in X coordinate";
+    ASSERT_FLOAT_EQ(int_points[1], 5.39535) << "FAIL in Y coordinate";
+}
+
+TEST(InformationEstimatesTests, MutualInformationTest)
+{
+    InformationEstimates inf_estimates;
+
+    karto::LocalizedRangeScan * s1 = new karto::LocalizedRangeScan();
+    karto::LocalizedRangeScan * s2 = new karto::LocalizedRangeScan();
```

Comment:
@kmilo7204 nit^2: consider using `std::unique_ptr` instances here. Better to be explicit about lifetime if we can afford it (and it seems so, I don't see any API taking ownership of these instances).